### PR TITLE
refactor: remove bloodbath and rework sanguinary

### PR DESCRIPTION
**Pull request by LordMidas**
- Sanguinary is now what Bloodbath used to be.
- Sanguinary now requires Cleaver.

Note: I did not test this in game as the changes are mostly copy pasting the Bloodbath working code into the Sanguinary perk with only minor tweaks to the code.

### CODE REVIEW

File: scripts/skills/perks/perk_rf_sanguinary.nut [hunk 1]
```diff
@@ -17,78 +18,35 @@ this.perk_rf_sanguinary <- ::inherit("scripts/skills/skill", {
...
-			this.m.IsHitHandlingComplete = true;
-		}
-	}
-
-	// Handles cases where Bleeding is applied by the attack e.g. during cleave skill's onUse function
-	// Note: Is unable to handle bleeding applied by "scheduledEvent" attacks
-	function onAnySkillExecuted( _skill, _targetTile, _targetEntity, _forFree )
-	{
-		if (this.m.IsHitHandlingComplete || !this.isAttackValid(_skill, this.getContainer().getActor(), _targetEntity))
+		if (_skill == null || !_skill.isAttack() || (!this.m.RequiresWeapon && !_skill.m.IsWeaponSkill) || !this.isEnabled())
```
Comment (Darxo) [inline scripts/skills/perks/perk_rf_sanguinary.nut:38]: I don't understand this line `(!this.m.RequiresWeapon && !_skill.m.IsWeaponSkill)`

If no weapon is required and the skill is not a weapon skill, then nothing happens?

I feel like the first `!` is wrong and should be removed.

File: scripts/skills/perks/perk_rf_sanguinary.nut [hunk 2]
```diff
@@ -1,8 +1,9 @@
 this.perk_rf_sanguinary <- ::inherit("scripts/skills/skill", {
 	m = {
-		FatigueCostRefundPercentage = 25,
-		IsHitHandlingComplete = true,
-		WasBleeding = false
+		IsForceEnabled = false,
+		RequiresWeapon = true,
```
Comment (Darxo) [inline scripts/skills/perks/perk_rf_sanguinary.nut:4]: Why exactly do we need `IsForceEnabled` aswell as `RequiresWeapon`? It seems like they do the same thing.
I would like comments explaining the purpose of both of them.

File: scripts/skills/perks/perk_rf_sanguinary.nut (hunk 1 above)
Comment (Darxo) [inline scripts/skills/perks/perk_rf_sanguinary.nut:38]: Even if you fix the issue with the `!` above.

There is still the issue that `IsForceEnabled` is not strong enough to force-enable this effect if the skill is not a weapon skill. In that case we abort (`return;`) before `IsForceEnabled` inside the `this.isEnabled()` can do its work.

What exactly is `IsForceEnabled` supposed to do (see remark at the top about comment in member declaration), if not circumventing the weapon restriction?

**Review by Darxo — CHANGES_REQUESTED** [review]

File: scripts/skills/perks/perk_rf_sanguinary.nut (hunk 1 above)
Comment (LordMidas) [reply scripts/skills/perks/perk_rf_sanguinary.nut]: This was the older system and quite confusing tbh to work with too. I've improved it and pushed a commit now using `isSkillValid` instead.

**Review by Darxo — APPROVED** [review]

**Pull request merged**